### PR TITLE
finished routes for user login/logout/registration

### PR DESCRIPTION
**Pull request by DaleTsakamoto**
Added routes to the routes/users.js file,
Created authorization in the auth.js file requireAuth,
Edited app.js to include app.use(restoreUser)

### CODE REVIEW

File: routes/users.js [hunk 1]
```diff
@@ -1,10 +1,104 @@
...
+    })
+  }
+}));
+
+/* GET user log-in. */
+router.get('/login', csrfProtection, (req, res) => {
+  res.render('log-in', {
+    token: req.csrfToken()
+  })
+})
```
Comment (scottgit) [inline routes/users.js]: I believe all the routes should be passing a `title` value to the page.

File: routes/users.js [hunk 2]
```diff
@@ -1,10 +1,104 @@
...
+    } else {
+      const user = await db.User.findOne({ where: { username } })
+      if (user !== null) {
+        const passwordMatch = await bcrypt.compare(password, user.hashedPassword.toString());
+        if (passwordMatch) {
+          loginUser(req, res, user)
+          return res.redirect('/')
+        }
+      }
+    }
```
Comment (scottgit) [inline routes/users.js]: To DRY up this code, set a `let user;` before the if block, just do the `if` to assign that user a value from the correct db call, then have one block of code do the `if(user!==null)`

**Review by scottgit — CHANGES_REQUESTED** [review]

File: routes/users.js (hunk 1 above)
Comment (DaleTsakamoto) [reply routes/users.js]: fixed

File: routes/users.js (hunk 2 above)
Comment (DaleTsakamoto) [reply routes/users.js]: fixed

**Review by scottgit — APPROVED** [review]
Those changes look good

**Pull request merged**